main.go

```diff
@@ -215,3 +221,30 @@ func ensureManagedOCS(c client.Client, log logr.Logger, envVars map[string]strin
 		return err
 	}
 }
+
+func checkIfCRDExist(crdNames []string) map[string]bool {
```

Comment:
A function prefixed `check` indicated a boolean result which is not the case.
I would suggest renaming the function to `mapCRDAvailability` which makes sense as the function return a map with a boolean indicating if each of the requested CRD is available or not